### ISSUE TITLE
[Power Up] Batch Single-Qubit Circuit Training for Image Classification

### ISSUE THREAD

**Issue by jkwn** [XanaduAI/QHack2021#44]
### Team Name: 

Qwerty

### Project Description: 

In this project, we explore the trainability of quantum circuits that consist of a single qubit or a small number of qubits (wires), for the image classification task of CIFAR-10 [1]. This is motivated from the following observations and thoughts.

1) A single-qubit circuit can be trained to represent a classification function with 3 classes. (We used a single-qubit circuit to solve the QML Challenge problem "Circuit Training 500".) With higher training accuracy, a single-qubit circuit could represent a function with 10 classes.

2) The input of colored image classification task generally has 3 channels, representing R, G, and B intensities of pixels. On the other hand, independent rotation operations can be performed on a qubit along 3 axes, namely, X, Y, and Z. Therefore, a pixel of an input image can be naturally encoded into a rotation on a qubit. 

3) With a small-number-of-qubit quantum circuit representing a classification function, the batch training could be done by running those circuits in parallel.

The plan is as follows:
1) Train a single-qubit circuit for the CIFAR-10 task. (partially done)
2) Train a 2-qubit circuit, by splitting the above single-qubit circuit into two parts, where the second one performs the reverse operations of the second half of the original single-qubit circuit.
3) Train a 4-qubit circuit.
4) Train a 8-qubit circuit.
5) Draw a graph showing the time-space tradeoff of quantum circuits for the same task.

### Source code: 

https://github.com/jkwn/Qwerty

### Resource Estimate: 

We are planning to use the IonQ device. 
Step 1) 100 tasks, 2000 shots ($300 + $200)
Step 2) 100 tasks, 2000 shots ($300 + $200)
Step 3) 100 tasks, 2000 shots ($300 + $200)
Step 4) 100 tasks, 2000 shots ($300 + $200)

[1] https://www.cs.toronto.edu/~kriz/cifar.html

**Comment by co9olguy** [comment]
Thanks for your Power Up Submission @jkwn !

To help us keep track of final submissions, we will be closing all of the [Power Up] issues. We ask you to open a new issue for your final submission. Please use [this pre-formatted [Entry] Issue template](https://github.com/XanaduAI/QHack/issues/new?assignees=&labels=submission&template=open-hackathon-final-submission.md&title=%5BENTRY%5D+Your+Project+Title). Note that for the final submission, the Resource Estimate requirement is replaced by a Presentation item.
